File: evd_ros_backend/evd_ros_core/src/evd_script/program/tasks.py/open_gripper.py
from ..task import Task
from ..primitives import Gripper


class OpenGripper(Task):

    '''
    Data structure methods
    '''

    def __init__(self, position=0, effort=100, speed=100, type='', name='',
                 uuid=None, parent=None, append_type=True, primitives=None, context=None):

        if primitives == None:
            primitives=[
                Gripper(
                    position=position,
                    effort=effort,
                    speed=speed)
            ]

        super(OpenGripper,self).__init__(
            type='open-gripper.'+type if append_type else type,
            name=name,
            uuid=uuid,
            parent=parent,
            append_type=append_type,
            context=context,
            primitives=primitives)

File: evd_authoring_app/src/model/evd_script/program/tasks/open_gripper.js
import { Task } from '../task';
import { Gripper } from '../primitive';
import { NodeParser } from '../../utility_functions';


export class OpenGripper extends Task {

    /*
    * Data structure methods
    */

    static typeString() {
        return 'open-gripper';
    }

    static fullTypeString() {
        return Task.fullTypeString() + OpenGripper.typeString();
    }

    constructor(position=0, effort=100, speed=100, thingUuid=null, type='', name='', uuid=null, parent=null, appendType=true, primitives=null) {

        if (primitives !== null) {
            primitives = [
                Gripper(
                    thingUuid= thingUuid,
                    position= position,
                    effort= effort,
                    speed= speed
                )
            ];
        }

        super(
            type= (appendType) ? 'open-gripper.'+type : type,
            name= name,
            uuid= uuid,
            parent= parent,
            appendType= appendType,
            primitives= primitives
        );
    }

    static fromDict(dct) {
        return OpenGripper(
            type= dct.type,
            name= dct.name,
            uuid= dct.uuid,
            appendType= false,
            primitives= dct.primitives.map(p => NodeParser(p))
        );
    }
}